client/app/dedicated/server/dedicated-server.service.js

```diff
@@ -687,18 +687,25 @@ angular
             });
         };
 
-        this.ipmiStartConnection = function (serviceName, type, ttl, ipToAllow, sshKey) {
-            return OvhHttp.post("/dedicated/server/{serviceName}/features/ipmi/access", {
+        this.ipmiStartConnection = function ({ serviceName, type, ttl, ipToAllow, sshKey, withGeolocation }) {
+            let promise = $q.when(ipToAllow);
+
+            if (withGeolocation) {
+                promise = this.getIpGeolocation().then((data) => _.get(data, "ip"));
+            }
+
+            return promise.then((ip) => OvhHttp.post("/dedicated/server/{serviceName}/features/ipmi/access", {
                 rootPath: "apiv6",
                 urlParams: {
                     serviceName
                 },
                 data: {
                     ttl,
                     type,
-                    sshKey
+                    sshKey,
+                    ipToAllow: ip
```

Comment:
If you param has been renamed (see previous comment), you can use shorthand:

```js
{
  …
  sshKey,
  ip
}
```

Reply:
error  'ipToAllow' is already declared in the upper scope  no-shadow